ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -233,8 +505,9 @@ module AuthADWinbind : Auth_signature.AUTH_MODULE = struct
       Raises auth_failure if authentication is not successful
   *)
 
-  let authenticate_username_password username password =
-    "authenticate_ticket To be implemented in CP-35399"
+  let authenticate_username_password uname password =
+    let () = maybe_raise (ntlm_auth uname password) in
```

Comment:
@liulinC the behaviour of ntlm_auth is really strange. I found that sometimes it will not accept the username in anything other than SAM format (that is, `echo "$PASSWD" | netlm_auth --username user` doesn't work, but `echo "$PASSWD" | ntlm_auth --username "DOMAIN\user"` does work)

I think we might need to convert `uname` to SAM format just in case

Reply:
Yes, you are right, this is an existing issue with PBIS, and we need to fix it. 
please update it in this PR if you like.